src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -439,32 +439,163 @@ class IndexLogEntryTest extends SparkFunSuite with SQLHelper with BeforeAndAfter
     FileUtils.deleteDirectory(tempDir.toFile)
   }
 
-  test("Directory.merge Test: test if merge works as expected.") {
-    // File Structure
-    // testDir/a/f1
-    // testDir/a/f2
-    // testDir/a/b/f3
-    // testDir/a/b/f4
+  test("Directory.merge: test if merge works as expected.") {
+    // directory1:
+    // a/f1
+    // a/f2
+    val directory1 = Directory(
+      name = "a",
+      files = Seq(
+        FileInfo("f1", 100L, 100L),
+        FileInfo("f2", 100L, 100L)
+      )
+    )
 
-    val tempDir = Files.createDirectories(Paths.get(testDir + "/temp"))
-    val a = Files.createDirectories(Paths.get(tempDir + "/a"))
-    val b = Files.createDirectories(Paths.get(a + "/b"))
-    val f1 = Files.createFile(Paths.get(a + "/f1"))
-    val f2 = Files.createFile(Paths.get(a + "/f2"))
-    val f3 = Files.createFile(Paths.get(b + "/f3"))
-    val f4 = Files.createFile(Paths.get(b + "/f4"))
+    // directory2:
+    // a/b/f3
+    // a/b/f4
+    val directory2 = Directory(
+      name = "a",
+      subDirs = Seq(
+        Directory(
+          name = "b",
+          files = Seq(
+            FileInfo("f3", 100L, 100L),
+            FileInfo("f4", 100L, 100L)
+          )
+        )
+      )
+    )
 
-    val aDirectory = Directory.fromDirectory(toPath(a))
-    val bDirectory = Directory.fromDirectory(toPath(b))
+    // Expected result of merging directory1 and directory2:
+    // a/f1
+    // a/f2
+    // a/b/f3
+    // a/b/f4
+    val expected = Directory(
+      name = "a",
+      files = Seq(
+        FileInfo("f1", 100L, 100L),
+        FileInfo("f2", 100L, 100L)
+      ),
+      subDirs = Seq(
+        Directory(
+          name = "b",
+          files = Seq(
+            FileInfo("f3", 100L, 100L),
+            FileInfo("f4", 100L, 100L)
+          )
+        )
+      )
+    )
 
-    val expected = Directory.fromLeafFiles(Seq(f1, f2, f3, f4).map(toFileStatus))
-    val actual1 = aDirectory.merge(bDirectory)
-    val actual2 = bDirectory.merge(aDirectory)
+    val actual1 = directory1.merge(directory2)
+    val actual2 = directory2.merge(directory1)
 
     assert(directoryEquals(actual1, expected))
     assert(directoryEquals(actual2, expected))
+  }
 
-    FileUtils.deleteDirectory(tempDir.toFile)
+  test("Directory.merge: test if merge works as expected when directories overlap.") {
+    // directory1:
+    // a/f1
+    // a/f2
+    // a/b/f3
```

Comment:
Thank you for all these examples in the comments! Makes it very easy to understand and cross-check if the test is testing as expected :)